src/Panel/Collector/FilesCollector.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Kirby\Panel\Collector;
+
+use Kirby\Cms\Files;
+use Kirby\Cms\Page;
+use Kirby\Cms\Pages;
+use Kirby\Cms\Site;
+use Kirby\Cms\User;
+use Kirby\Cms\Users;
+
+class FilesCollector extends ModelsCollector
+{
+	public function __construct(
+		protected Site|Page|User $parent,
+		protected bool $flip = false,
+		protected int|null $limit = null,
+		protected int $page = 1,
+		protected string|null $query = null,
+		protected string|null $search = null,
+		protected string|null $sortBy = null,
+		protected string|null $template = null,
+	) {
+	}
+
+	protected function collectByParent(): Files
+	{
+		return $this->parent->files();
+	}
+
+	protected function collectByQuery(): Files
+	{
+		return $this->parent->query($this->query, Files::class) ?? new Files([]);
+	}
```

Comment:
`Discussion:` Wondering if we would rather define the Model class (here `Files::class`) as e.g. a class constant and then the individual model collector classes wouldn't have to repeat this method with almost the same code.

Reply:
I've tried it but since the UsersCollector needs a adjusted version anyway, it feels a bit too abstracted for two classes.